patches/v2.15.1.patch

```diff
@@ -0,0 +1,123 @@
+--- a/src/app/app.component.ts
++++ b/src/app/app.component.ts
+@@ -140,6 +140,10 @@ export class AppComponent implements OnDestroy, OnInit {
+                         }
+                         break;
+                     case 'showToast':
++                        if (typeof message.text === "string" && (message.text.indexOf("this.subtle") != -1 || message.text.indexOf("importKey") != -1)) {
++                            message.title="This browser requires HTTPS to use the web vault";
++                            message.text="Check the bitwarden_rs wiki for details on how to enable it";
++                        }
+                         this.showToast(message);
+                         break;
+                     case 'analyticsEventTrack':
+diff --git a/src/app/services/services.module.ts b/src/app/services/services.module.ts
+index e64240ad..025fdef1 100644
+--- a/src/app/services/services.module.ts
++++ b/src/app/services/services.module.ts
+@@ -127,24 +127,32 @@ const environmentService = new EnvironmentService(apiService, storageService, no
+ const auditService = new AuditService(cryptoFunctionService, apiService);
+ const eventLoggingService = new EventLoggingService(storageService, apiService, userService, cipherService);
+ 
+-const analytics = new Analytics(window, () => platformUtilsService.isDev() || platformUtilsService.isSelfHost(),
++const analytics = new Analytics(window, () => platformUtilsService.isDev() || platformUtilsService.isSelfHost() || true,
+     platformUtilsService, storageService, appIdService);
+ containerService.attachToWindow(window);
+ 
+ export function initFactory(): Function {
++    function getBaseUrl() {
++        // If the base URL is `https://bitwarden.example.com/base/path/`,
++        // `window.location.href` should have one of the following forms:
++        //
++        // - `https://bitwarden.example.com/base/path/`
++        // - `https://bitwarden.example.com/base/path/#/some/route[?queryParam=...]`
++        //
++        // We want to get to just `https://bitwarden.example.com/base/path`.
++        let baseUrl = window.location.href;
++        baseUrl = baseUrl.replace(/#.*/, '');  // Strip off `#` and everything after.
++        baseUrl = baseUrl.replace(/\/+$/, ''); // Trim any trailing `/` chars.
++        return baseUrl;
++    }
+     return async () => {
+         await (storageService as HtmlStorageService).init();
+-        const isDev = platformUtilsService.isDev();
+-        if (!isDev && platformUtilsService.isSelfHost()) {
+-            environmentService.baseUrl = window.location.origin;
+-        } else {
+-            environmentService.notificationsUrl = isDev ? 'http://localhost:61840' :
+-                'https://notifications.bitwarden.com'; // window.location.origin + '/notifications';
+-            environmentService.enterpriseUrl = isDev ? 'http://localhost:61840' :
+-                'https://enterprise.bitwarden.com'; // window.location.origin + '/enterprise';
```

Comment:
The main difference from upstream is the addition of these `enterpriseUrl` lines.